src/nni_manager/core/nnimanager.ts

```diff
@@ -131,19 +129,34 @@ class NNIManager implements Manager {
         return this.dataStore.exportTrialHpConfigs();
     }
 
-    public addCustomizedTrialJob(hyperParams: string): Promise<void> {
+    public addCustomizedTrialJob(hyperParams: string): Promise<number> {
         if (this.readonly) {
             return Promise.reject(new Error('Error: can not add customized trial job in readonly mode!'));
         }
         if (this.currSubmittedTrialNum >= this.experimentProfile.params.maxTrialNum) {
-            return Promise.reject(
-                new Error('reach maxTrialNum')
-            );
+            return Promise.reject(new Error('reach maxTrialNum'));
         }
-        this.customizedTrials.push(hyperParams);
+
+        // TODO: NNI manager should not peek tuner's internal protocol, let's refactor this later
```

Comment:
any idea about how to refactor?

Reply:
multi-phase does not work in this implementation. but if the customized configuration is sent to tuner first, then multi-phase can be supported

Reply:
The refactor should be in dispatcher protocol. I'll do it after I understand the protocol clearly.
To me multi-phase is also a protocol problem so I don't know how to do it for now.